services/brig/src/Brig/Team/API.hs

```diff
@@ -219,15 +232,28 @@ createInvitation' ::
   EmailAddress ->
   Public.InvitationRequest ->
   Handler r (Public.Invitation, Public.InvitationCode)
-createInvitation' tid mUid inviteeRole mbInviterUid fromEmail body = do
-  let email = (inviteeEmail body)
+createInvitation' tid mUid inviteeRole mbInviterUid fromEmail invRequest = do
+  let email = inviteeEmail invRequest
   let uke = mkEmailKey email
   blacklistedEm <- lift $ liftSem $ isBlocked email
   when blacklistedEm $
     throwStd blacklistedEmail
   emailTaken <- lift $ liftSem $ isJust <$> lookupKey uke
+  migratePersonalUser <-
+    if emailTaken
+      then do
+        mAccount <- lift $ liftSem $ getLocalUserAccountByUserKey =<< qualifyLocal' uke
+        pure $ case mAccount of
+          -- this can e.g. happen if the key is claimed but the account is not yet created
+          Nothing -> False
+          Just account ->
+            account.accountStatus == Active
+              && isNothing account.accountUser.userTeam
+      else pure False
+
   when emailTaken $
-    throwStd emailExists
+    unless migratePersonalUser $
```

Comment:
if you define `migratePersonalUser` inside the `when emailTaken` block, you can drop the `if emailTaken` wrapping.